contracts/src/tokens/memecoin.cairo

```diff
@@ -148,8 +150,9 @@ mod UnruggableMemecoin {
         }
 
         fn transfer(ref self: ContractState, recipient: ContractAddress, amount: u256) -> bool {
-            self._check_max_buy_percentage(amount);
             let sender = get_caller_address();
+            self._check_and_update_tx_hash(recipient);
```

Comment:
The destination of a swap can be someone else other than the caller/recipient. In that case, if I let's say Alice, do the following transactoin
- Swap from alice to: Alice: ✅ 
- Swap from Alice to: Bob: ✅ 
- Swap from Alice to Alice: 🛑 

If I make a multicall with multiple swaps and multiple recipients, I can still make multiple swaps in a single tx.